lib/query.js

```diff
@@ -45,6 +51,44 @@ export const isNumber = node => {
     }
 }
 
+export const isInteger = node => {
+    return isNumber(node) && Number.isInteger(Number(node.value))
+}
+
+export const isPolynomial = (node) => {
+    return isAdd(node) && node.args.every(isPolynomialTerm)
+}
+
+export const isVariableFactor = (node) =>
+    isIdentifier(node) ||
+    isPow(node) && isIdentifier(node.args[0])
+    && (isNumber(node.args[1]) || isVariableFactor(node.args[1]))
+
+export const isPolynomialTerm = (node) => {
+    if (isNumber(node) || isConstantFraction(node) || isDecimal(node)) {
+        return true
+    } else if (isIdentifier(node)) {
+        return true
+    } else if (isPow(node)) {
+        const [base, exponent] = node.args
+        return (isIdentifier(base) || isPolynomial(base)) && isPolynomialTerm(exponent)
```

Comment:
I think it does, somehow. I tried making the change and it altered some of the snapshots.